replica/database.cc

```diff
@@ -1545,7 +1545,9 @@ database::query(schema_ptr s, const query::read_command& cmd, query::result_opti
     std::exception_ptr ex;
 
     if (cmd.query_uuid && !cmd.is_first_page) {
-        querier_opt = _querier_cache.lookup_data_querier(cmd.query_uuid, *s, ranges.front(), cmd.slice, semaphore, trace_state, timeout);
+        if (auto found = _querier_cache.lookup_data_querier(cmd.query_uuid, *s, ranges.front(), cmd.slice, semaphore, trace_state, timeout)) {
+            querier_opt.emplace(std::move(*found));
+        }
```

Comment:
How does this improve anything? Usually lookup_data_querier returns nullopt and operator= will do almost nothing. Moreover, the optimizer should remember that querier_opt is disengaged.

Reply:
I agree, this just makes the code slightly harder to read, without any apparent gain.

Reply:
true, it has very little effect.
I needed it at some point in development when the copy constructor was deleted.
But it's not a must to have.